packages/react-router/src/link.tsx

```diff
@@ -74,46 +74,53 @@ export type RelativeToPathAutoComplete<
   AllPaths extends string,
   TFrom extends string,
   TTo extends string,
-  SplitPaths extends string[] = Split<AllPaths, false>,
 > = TTo extends `..${infer _}`
-  ? SplitPaths extends [
-      ...Split<ResolveRelativePath<TFrom, TTo>, false>,
-      ...infer TToRest,
-    ]
-    ? `${CleanPath<
-        Join<
-          [
-            ...Split<TTo, false>,
-            ...(
-              | TToRest
-              | (Split<
-                  ResolveRelativePath<TFrom, TTo>,
-                  false
-                >['length'] extends 1
-                  ? never
-                  : ['../'])
-            ),
-          ]
-        >
-      >}`
-    : never
-  : TTo extends `./${infer RestTTo}`
+  ? Split<AllPaths, false> extends infer SplitPaths
     ? SplitPaths extends [
-        ...Split<TFrom, false>,
-        ...Split<RestTTo, false>,
-        ...infer RestPath,
+        ...Split<ResolveRelativePath<TFrom, TTo>, false>,
+        ...infer TToRest,
       ]
-      ? `${TTo}${Join<RestPath>}`
+      ? `${CleanPath<
+          Join<
+            [
+              ...Split<TTo, false>,
+              ...(
+                | TToRest
+                | (Split<
+                    ResolveRelativePath<TFrom, TTo>,
+                    false
+                  >['length'] extends 1
+                    ? never
+                    : ['../'])
+              ),
+            ]
+          >
+        >}`
+      : never
+    : never
+  : TTo extends `./${infer RestTTo}`
+    ? Split<AllPaths, false> extends infer SplitPaths
+      ? SplitPaths extends [
+          ...Split<TFrom, false>,
+          ...Split<RestTTo, false>,
+          ...infer RestPath,
+        ]
+        ? `${TTo}${Join<RestPath>}`
+        : never
       : never
     :
-        | (TFrom extends `/`
-            ? never
-            : SplitPaths extends [...Split<TFrom, false>, ...infer RestPath]
-              ? Join<RestPath> extends { length: 0 }
-                ? never
-                : './'
-              : never)
-        | (TFrom extends `/` ? never : '../')
+        | (string extends TFrom
```

Comment:
string is the current default for Link so I had to check for this. But I don't know if it should be '/'